File: src/main/java/me/devsnox/varoxtime/listeners/PlayerListener.java
package me.devsnox.varoxtime.listeners;

import me.devsnox.varoxtime.varoxtime.TimeManager;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import org.bukkit.event.player.PlayerQuitEvent;

/**
 * Copyright by DevSnox
 * E-Mail: me.devsnox@gmail.com
 * Skype: DevSnox
 */
public class PlayerListener implements Listener {

    private TimeManager timeManager;

    public PlayerListener(TimeManager timeManager) {
        this.timeManager = timeManager;
    }

    @EventHandler
    public void onJoin(PlayerJoinEvent event) {
        Player player = event.getPlayer();
        if(this.timeManager.exists(player.getUniqueId())) {
            this.timeManager.loadPlayer(player.getUniqueId());
        } else {
            this.timeManager.createPlayer(player.getUniqueId());
            this.timeManager.loadPlayer(player.getUniqueId());
        }
    }

    @EventHandler
    public void onQuit(PlayerQuitEvent event) {
        Player player = event.getPlayer();
        if(this.timeManager.isLoaded(player.getUniqueId())) {
            this.timeManager.unloadPlayer(player.getUniqueId());
        }
    }
}

File: src/main/java/me/devsnox/playtime/listeners/PlayerListener.kt
package me.devsnox.playtime.listeners

import me.devsnox.playtime.playtime.TimeManager
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent
import org.bukkit.event.player.PlayerQuitEvent

/**
 * Created by Yasin Dalal
 * Copyright (c) 2017-2019 Yasin Dalal
 * GitHub: https://github.com/DevSnox
 * E-Mail: yasin@dalal.ch
 */
class PlayerListener(private val timeManager: TimeManager) : Listener {

    @EventHandler
    fun onJoin(event: PlayerJoinEvent) {
        val uuid = event.player.uniqueId
        if (!timeManager.exists(uuid)) timeManager.createPlayer(uuid)
        timeManager.loadPlayer(uuid)
    }

    @EventHandler
    fun onQuit(event: PlayerQuitEvent) {
        val uuid = event.player.uniqueId
        if (!timeManager.isLoaded(uuid)) return
        timeManager.unloadPlayer(uuid)
    }

}